src/workerd/api/urlpattern-standard.c++

```diff
@@ -0,0 +1,253 @@
+#define V(_, name)                                                                                 \
+  KJ_IF_SOME(v, name) {                                                                            \
+    init.name = std::string(v.cStr(), v.size());                                                   \
+  }
+  URL_PATTERN_COMPONENTS(V)
+#undef V
+  KJ_IF_SOME(b, baseURL) {
+    init.base_url = std::string(b.cStr(), b.size());
+  }
+  return init;
+}
+
+URLPattern::URLPatternInit URLPattern::createURLPatternInit(const ada::url_pattern_init& other) {
+  URLPatternInit result{};
+#define V(_, name)                                                                                 \
+  if (auto v = other.name) {                                                                       \
+    result.name = kj::str(kj::ArrayPtr(v->c_str(), v->size()));                                    \
+  }
+  URL_PATTERN_COMPONENTS(V)
+#undef V
+
+  if (auto v = other.base_url) {
+    result.baseURL = kj::str(kj::ArrayPtr(v->c_str(), v->size()));
+  }
+  return result;
+}
+
+URLPattern::URLPatternComponentResult URLPattern::createURLPatternComponentResult(
+    jsg::Lock& js, const ada::url_pattern_component_result& other) {
+  auto result = URLPatternComponentResult{
+    .input = js.str(kj::ArrayPtr(other.input.c_str(), other.input.size())),
+    .groups = js.obj(),
+  };
+
+  for (auto& [key, value]: other.groups) {
+    auto k = js.str(kj::ArrayPtr(key.c_str(), key.size()));
+
+    if (value) {
+      result.groups.set(js, k, js.str(kj::ArrayPtr(value->c_str(), value->size())));
+    } else {
+      result.groups.set(js, k, js.undefined());
+    }
+  }
+  return result;
+}
+
+bool URLPattern::getHasRegExpGroups() const {
+  return inner.has_regexp_groups();
+}
+
+URLPattern::URLPatternResult URLPattern::createURLPatternResult(
+    jsg::Lock& js, const ada::url_pattern_result& other) {
+  URLPatternResult result{
+#define V(_, name) .name = URLPattern::createURLPatternComponentResult(js, other.name),
+    URL_PATTERN_COMPONENTS(V)
+#undef V
+  };
+
+  auto vecInputs = kj::heapArray<kj::OneOf<jsg::JsString, URLPatternInit>>(other.inputs.size());
+  size_t i = 0;
+  for (const auto& input: other.inputs) {
+    if (std::holds_alternative<std::string_view>(input)) {
+      auto raw = std::get<std::string_view>(input);
+      vecInputs[i] = js.str(kj::ArrayPtr(raw.data(), raw.size()));
+    } else {
+      KJ_DASSERT(std::holds_alternative<ada::url_pattern_init>(input));
+      auto obj = std::get<ada::url_pattern_init>(input);
+      vecInputs[i] = createURLPatternInit(obj);
+    }
+    i++;
+  }
+  result.inputs = mv(vecInputs);
+  return result;
+}
+
+#define DEFINE_GETTER(uppercase, lowercase)                                                        \
+  kj::StringPtr URLPattern::get##uppercase() const {                                               \
+    auto value = inner.get_##lowercase();                                                          \
```

Comment:
Is value an owned object here? is it a string_view?
Just gotta make sure that the underlying string outlives the return StringPtr.

Reply:
It is std::string_view where the underlying data is owned by ada::url_pattern class